### PR TITLE
test: add comprehensive unit tests for streak calculation utilities

### LINKED ISSUE #1069: test: add unit tests for streak calculation utility functions
## Testing Request

**What tests are missing?**
The streak calculation logic (current streak, longest streak, streak at risk) is a core feature of DevTrack but appears to lack comprehensive unit tests. These functions are business-critical — a bug here directly impacts user trust.

**Proposed Tests**
```typescript
describe('streak calculations', () => {
  it('returns 0 for empty commit history')
  it('counts consecutive days correctly')
  it('resets streak on missed day')
  it('handles streak freeze correctly (frozen day should not break streak)')
  it('calculates longest streak from history')
  it('detects streak at risk (no commit today)')
  it('handles timezone boundaries correctly')
  it('counts today correctly if commit made after midnight')
  it('handles leap year Feb 29 correctly')
  it('handles months with 28, 29, 30, 31 days')
})
```

**Implementation Details**
- Find streak utility functions in `src/lib/` or similar
- Create `__tests__/streak.test.ts`
- Pure unit tests — no DB or API mocking needed
- Test edge cases: year boundaries, timezone changes, DST

**Acceptance Criteria**
- [ ] All 10 test cases implemented
- [ ] 100% branch coverage for streak calculation functions
- [ ] Tests run in < 100ms (pure unit tests)
- [ ] Edge cases (timezones, leap years) covered

### PR DESCRIPTION
**Pull request by Shanidhya01**
## Summary

Adds comprehensive unit tests for streak calculation utility functions to improve reliability and prevent regressions in DevTrack's streak tracking system.

Closes #1069

## Changes Made

* Added unit tests for current streak calculation
* Added unit tests for longest streak calculation
* Added unit tests for streak-at-risk detection
* Added tests for streak reset behavior
* Added tests for empty commit history
* Added tests for date boundary edge cases
* Added tests for leap year handling
* Added tests for month transitions (28/29/30/31 days)
* Added tests for timezone-related scenarios
* Added tests for commits made after midnight

## Test Coverage

Covered scenarios:

* Returns 0 for empty commit history
* Counts consecutive days correctly
* Resets streak on missed day
* Handles streak freeze functionality (if implemented)
* Calculates longest streak accurately
* Detects streak at risk when no commit exists today
* Handles timezone boundaries correctly
* Counts today correctly for post-midnight commits
* Handles leap year dates (Feb 29)
* Handles month length variations

## Type of Change

* [x] Test addition
* [ ] Bug fix
* [ ] New feature
* [ ] Breaking change

## Checklist

* [x] Tests added
* [x] Existing tests pass
* [x] No production behavior changed
* [x] Code follows project conventions

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -51,6 +51,7 @@
     "react-markdown": "^10.1.0",
     "recharts": "^2.15.4",
     "rehype-sanitize": "^6.0.0",
+    "rolldown": "^1.0.3",
```
Comment (Legit-Ox) [inline package.json]: **Wrong section:** `rolldown` is a Rust-based bundler (build tool), not a runtime library. It belongs in `devDependencies` at most — but it's not needed here at all. The test file only imports from `@/lib/streak` and `@/lib/streak-utils`; the project already uses Vitest for running tests. Remove this entirely. Shipping rolldown as a prod dependency adds ~50MB to the install and is unrelated to this PR.

**Review by Legit-Ox — COMMENTED** [review]
Test coverage is solid — good edge case variety across leap years, month transitions, timezone boundaries, and freeze dates. Three issues before merging:

1. `rolldown` added to production `dependencies` — unrelated to this PR and shouldn't be there at all
2. One test name directly contradicts its assertion
3. `package-lock.json` accidentally narrows the Node engine range

File: test/streak.test.ts [hunk 2]
```diff
@@ -0,0 +1,614 @@
...
+        "2026-06-10",
+        "2026-06-11",
+        "2026-06-13",
+        "2026-06-14",
+        "2026-06-15",
+      ])
+    ).toBe(3);
+  });
+
+  it("deduplicates multiple commits on the same day", () => {
```
Comment (Legit-Ox) [inline test/streak.test.ts:436]: Test name contradicts the assertion. The title says `"does not treat Feb 28 → Mar 1 as consecutive"` but the assertion is `expect(result.longest).toBe(2)` — which proves they ARE consecutive. The inline comment even says `// so these ARE consecutive`. In a non-leap year 2025, Feb 28 → Mar 1 is a 1-day gap, so consecutive is correct — the name is just wrong.

Rename to: `"treats Feb 28 → Mar 1 as consecutive in a non-leap year (28-day February)"`

File: package-lock.json [hunk 3]
```diff
@@ -81,7 +82,7 @@
         "web-tree-sitter": "^0.26.9"
       },
       "engines": {
-        "node": ">=20.0.0"
+        "node": "20.x"
```
Comment (Legit-Ox) [inline package-lock.json]: Accidental narrowing: changed from `">=20.0.0"` to `"20.x"`. This excludes Node 22 and 24, which CI and Vercel run on. Looks like a side-effect of running `npm install` locally on Node 20. Revert to `">=20.0.0"`.

**Comment by Shanidhya01** [comment]
Ok will fix it by the end of the day

**Comment by Priyanshu-byte-coder** [comment]
The test coverage here is genuinely strong — the timezone edge cases, leap year handling, and streak-utils coverage are valuable. A few things to fix before merge:

1. **Remove the package.json change** — changing engines from 20.x to >=20.0.0 is unrelated to tests and belongs in a separate PR.

2. **package-lock.json changes** — the added resolved/integrity fields for rolldown are also unrelated. Please drop these.

3. **File conflict with #2776** — both PRs create test/streak.test.ts. Since #2776 has already been merged, please either rename your file (e.g. test/streak-comprehensive.test.ts) or consolidate the tests into the existing file to avoid conflicts.

Fix these and the test quality is solid enough to merge.

**Review by Priyanshu-byte-coder — APPROVED** [review]
Excellent comprehensive streak tests — timezone boundaries, leap years, freeze dates, edge cases. Well structured.
